client/src/js/services/grid/GridGrouping.js

```diff
@@ -180,10 +180,30 @@ function GridGroupingService(uiGridGroupingConstants, $filter, Session, $timeout
     return groupingDetail.grouping[0].colName;
   }
 
-  /** return back the list of selected rows **/
+  /**
+   * return back the list of selected rows,
+   * if one row is selected so all transaction row will be considered as selected
+   * TO DO : make it generic, any kind of group should used, not only trans_id
+   **/
+  function getSelectedGroups (){
+
+    var parsed = [], processedTransactions = [];
```

Comment:
Thanks, 
I know the reduce function and I already used that, but sometime the ```for loop``` structure seems more natural and clear for me. This point of view is very personal and I can be wrong.